internal/step/plugin/provider.go

```diff
@@ -890,6 +890,7 @@ func (r *runningStep) startStage(container deployer.Plugin) error {
 	previousStage := string(r.currentStage)
 	r.currentStage = StageIDStarting
 	inputRecievedEarly := false
+	r.atpClient = atp.NewClientWithLogger(container, r.logger)
```

Comment:
We're now calling `atp.NewClientWithLogger()` while holding `r.lock`.  We might not want to do that.

It might be better to call `atp.NewClientWithLogger()` before locking the lock, caching its return value in a local variable, and then assigning the value to `r.atpClient` after locking the lock.

Reply:
I see the merit of this. Done.